src/ml/neural_net/style_transfer/mps_style_transfer.m

```diff
@@ -251,6 +259,24 @@ - (instancetype) initWithDev:(id<MTLDevice>) dev
   return self;
 }
 
++ (NSData *) topLeftCropImage:(NSData *) data
+                  inputHeight:(NSUInteger) inputHeight
+                   inputWidth:(NSUInteger) inputWidth
+                 outputHeight:(NSUInteger) outputHeight
+                  outputWidth:(NSUInteger) outputWidth 
+                  numChannels:(NSUInteger) numChannels {
+  NSUInteger dataSize = sizeof(float) * numChannels * outputWidth * outputHeight;
+  NSMutableData *mutableData = [NSMutableData dataWithLength:dataSize];
+  for (NSUInteger idx = 0; idx < outputHeight; idx = idx + 1) {
+    NSUInteger rowSize = sizeof(float) * numChannels * outputWidth;
+    NSUInteger rowOffset = rowSize * idx;
+    NSUInteger startIndex = sizeof(float) * idx * numChannels * inputWidth;
+    NSRange range = NSMakeRange(startIndex, rowSize);
+    [data getBytes:mutableData.mutableBytes + rowOffset range:range];
+  }
```

Comment:
Let's try to adopt more of the patterns we see in e.g. NumPy and other various multi-array types we have, as well as in Apple APIs. Think in terms of strides. Something closer to:
```
NSUInteger outputRowStride = sizeof(float) * numChannels * outputWidth;
NSUInteger inputRowStride = sizeof(float) * numChannels * inputWidth;
for (NSUInteger idx = 0; idx < outputHeight; idx = idx + 1) {
  NSRange range = NSMakeRange(inputRowStride * idx, outputRowStride);
  [data getBytes:mutableData.mutableBytes + outputRowStride * idx range:range];
}
```
The above code should match the behavior of what you have here. But just to be clear, the desired behavior is to exclude only pixels on the bottom and right?

By reasoning about strides more explicitly here, we'll make it easier to reduce this to zero copies once `float_array` supports strides too. As an even more general principle, tend to factor out of loops variables than can be computed once beforehand.